cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceLifecycleContext.java

```diff
@@ -324,4 +326,10 @@ public MessageFetcher getMessageFetcher() {
     // TODO: CDAP-7807
     throw new UnsupportedOperationException("Messaging is not supported in MapReduce task-level context");
   }
+
+  @Override
+  public void record(Collection<? extends Operation> operations) {
```

Comment:
interesting: further up in this class we have a TODO to remove methods that are not supported in a task, and methods that emit a warning. Here, we throw an exception. Shouldn't it behave the same way for all unsupported operations?

Reply:
I think its better to throw an exception, since it gives the developer early notification to correct the code. TODO above is tracking the task where we want to support TMS in tasks, since we are not gonna support the recording of lineage in tasks, there is no need to have TODO.

Ideally `getHadoopJob` would also throw an exception, however changing that now means API behavior change. right?

Reply:
yes, ok